Modules/ScormAicc/classes/SCORM/class.ilSCORMItemGUI.php

```diff
@@ -61,13 +61,13 @@ function view()
 					? "?".$this->sc_object->getParameters()
 					: "";
 
-				$this->tpl = new ilTemplate("tpl.scorm_content_frameset.html", true, true, "Modules/ScormAicc");
+				$this->tpl = new ilGlobalTemplate("tpl.scorm_content_frameset.html", true, true, "Modules/ScormAicc");
```

Comment:
This is one of these many cases I currently do not understand. Why is this using ilGlobalTemplate?

Reply:
`$this->tpl->printToStdout` is uses some lines below.

Reply:
So this would hint at creating another copy of the `ilGlobalTemplate` for SCORM, right?

The list of the templates is this (via `grep -r "<html" . | grep "tpl"`, with purged `libs/*` files):

```
./templates/default/tpl.frameset.html:<html xmlns="http://www.w3.org/1999/xhtml">
./templates/default/tpl.main.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./setup/templates/tpl.error_page.html:<html lang="en">
./setup/templates/tpl.main.html:<html lang="en">
./Services/Database/templates/default/tpl.db_overview.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/Mail/templates/default/tpl.html_mail_template.html:<html xmlns="http://www.w3.org/1999/xhtml" xmlns="http://www.w3.org/1999/xhtml">
./Services/Mail/templates/default/tpl.mail_print.html:<html>
./Services/Container/templates/default/tpl.container_link_help.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/COPage/templates/default/tpl.fullscreen.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" class="ilc_Fullscreen">
./Services/Link/templates/default/tpl.link_help.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/RTE/tiny_mce_3_4_7/plugins/ibrowser/tpl.imagemanager.html:<html>
./Services/RTE/tiny_mce_3_4_7/plugins/ilimgupload/tpl.img_upload.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/RTE/tiny_mce_3_4_7/plugins/latex/tpl.latex.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/RTE/tiny_mce_3_5_11/plugins/ibrowser/tpl.imagemanager.html:<html>
./Services/RTE/tiny_mce_3_5_11/plugins/ilimgupload/tpl.img_upload.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/RTE/tiny_mce_3_5_11/plugins/latex/tpl.latex.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Services/LTI/templates/default/tpl.main.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Services/LTI/templates/default/tpl.lti_exit.html:<html>
./Modules/Test/templates/default/tpl.il_as_tst_print_body.html:<html>
./Modules/LearningModule/templates/default/tpl.fullscreen.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}" class="ilc_Fullscreen">
./Modules/LearningModule/templates/default/tpl.frameset.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/LearningModule/templates/default/tpl.blank.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.glossary_term_output.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.print.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.page_fullscreen.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.ppt_fullscreen.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.page_new.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.toc.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/LearningModule/templates/default/tpl.page.html:<html lang="{META_CONTENT_LANGUAGE}" dir="{LANGUAGE_DIRECTION}">
./Modules/Forum/templates/default/tpl.forums_export_print.html:<html>
./Modules/Forum/templates/default/tpl.forums_export_html.html:<html>
./Modules/Scorm2004/templates/default/tpl.scorm2004.player.html:<html xmlns="http://www.w3.org/1999/xhtml" id="scormplayer">
./Modules/Scorm2004/templates/default/tpl.scorm2004.specialpages.html:<html xmlns="http://www.w3.org/1999/xhtml" id="scormplayer">
./Modules/Scorm2004/templates/default/tpl.sco.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
./Modules/Scorm2004/templates/default/tpl.scorm2004.debug.html:<html lang="en">
./Modules/Scorm2004/templates/default/tpl.scorm2004.debugtxt.txt:<html>
./Modules/ScormAicc/templates/default/tpl.sahs_api.html:<html>
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset_one_page.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.som.html:<html>
./Modules/ScormAicc/templates/default/tpl.sahs_launch_cbt.html:<html>
./Modules/ScormAicc/templates/default/tpl.scorm_launch_asset.html:<html>
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset_js_one_page.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset_js.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.player12.html:<html>
./Modules/ScormAicc/templates/default/tpl.scorm_content_select.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.sahs_finish_cbt.html:<html>
./Modules/ScormAicc/templates/default/tpl.scorm_content_frameset.html:<html>
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.cache.html:<html manifest="{APPCACHE_URL}">
./Modules/ScormAicc/templates/default/tpl.sahs_exp_main.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Modules/ScormAicc/templates/default/tpl.sahs_edit_frameset.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.sahs_unload_cbt.html:<html>
./Modules/ScormAicc/templates/default/tpl.player2004.html:<html xmlns="http://www.w3.org/1999/xhtml" id="scormplayer">
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset_js_debug_one_page.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/ScormAicc/templates/default/tpl.sahs_pres_frameset_js_debug.html:<html xmlns="http://www.w3.org/1999/xhtml">
./Modules/Course/templates/default/tpl.crs_export.html:<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
./Modules/Chatroom/templates/default/tpl.history_export.html:<html>
```

Reply:
> So this would hint at creating another copy of the ilGlobalTemplate for SCORM, right?

No, if something does not make any use of ilGlobalTemplate features this would be a hint to use ilTemplate instead.

Reply:
Ok, we might be able to get away with using `ilTemplate` and `echo $this->tpl->get()` instead of using `printToStdout` here. But `printToStdout` is in fact a huge dependency on ilGlobalTemplate-features, because it uses all the `fill*`-methods, which in turn rely on variables in the template, other globals, etc.. Maybe @Uwe-Kohnle can help here.